python/cudf/cudf/_lib/cudf.pyx

```diff
@@ -176,12 +176,14 @@ cdef gdf_scalar* gdf_scalar_from_scalar(val, dtype=None) except? NULL:
     cdef gdf_scalar* s = <gdf_scalar*>malloc(sizeof(gdf_scalar))
     if s is NULL:
         raise MemoryError
+    if val is None:
+        is_valid = False
+        val = np.bool_(False)
```

Comment:
This will always make the `gdf_scalar` a bool or int8 (not sure which), and likely trigger a JIT op instead of a pre-compiled op. We should detect if we're passing `None` before this function and pass the `lhs.dtype` to the dtype parameter here and do something like: `val = dtype.type(0)`

Reply:
updated